extensions/sql/data/SQLIdentifier.h

```diff
@@ -19,18 +19,40 @@
 #pragma once
 
 #include <string>
-#include <vector>
+#include <functional>
 
-namespace org {
-namespace apache {
-namespace nifi {
-namespace minifi {
-namespace utils {
+namespace org::apache::nifi::minifi::sql {
 
-std::vector<std::string> inputStringToList(const std::string& str);
+class SQLIdentifier {
```

Comment:
I suggest renaming this to something like `SqlColumnName` to avoid confusion. "Identifier" typically refers to a primary key in the relational database context. "Column" is not the most precise term (that would be "attribute"), but it's the most intuitive.
And I suggest avoiding full-caps acronyms in CamelCase, because it makes it more difficult to determine word boundaries (e.g. `IXMLHTTPRequest`) and to programmatically convert between conventions (e.g. `s_q_l_identifier` instead of `sql_identifier`). It's also recommended in the [google style guide](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules).

Reply:
a quick googling seems to suggest [otherwise](https://www.google.com/search?q=sql+identifier)

Reply:
as for the capitalization, all other SQL entities in this extension have all-caps "SQL" in them, so I would stick to it for consistency

Reply:
fair. What do you think about `SQLColumnIdentifier`? Or do you intend this to be applicable to table and other identifiers as well?

Reply:
I think we can constrain it to columns for now, renamed it to `SQLColumnIdentifier` (if we need to use something like this for tables/others we can revisit this)